app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/miband/MiBand2Support.java

```diff
@@ -104,12 +108,25 @@ public void onReceive(Context context, Intent intent) {
 
             } else if (s.equals(BatteryInfoProfile.ACTION_BATTERY_INFO)) {
                 handleBatteryInfo((nodomain.freeyourgadget.gadgetbridge.service.btle.profiles.battery.BatteryInfo) intent.getParcelableExtra(BatteryInfoProfile.EXTRA_BATTERY_INFO));
+            } else if (s.equals(DeviceService.ACTION_MIBAND2_AUTH)) {
+                byte[] response = intent.getExtras().getByteArray(DeviceService.EXTRA_MIBAND2_AUTH_BYTE);
+                BluetoothGattCharacteristic temp = getCharacteristic(MiBand2Service.UUID_UNKNOWN_CHARACTERISTIC9);
+                temp.setValue(response);
+                if ((temp.getProperties() & (BluetoothGattCharacteristic.PROPERTY_WRITE | BluetoothGattCharacteristic.PROPERTY_WRITE_NO_RESPONSE)) != 0) {
+                    Log.d("HELLO", "its looks ok");
+                }
+                if (!mBluetoothGatt.writeCharacteristic(temp)) {
+                    Log.d("ERROR", "ERROR");
+                }
+                Log.d("HELLO", "HELLO");
+
             }
         }
     };
 
     private volatile boolean telephoneRinging;
     private volatile boolean isLocatingDevice;
+    private BluetoothGatt mBluetoothGatt;
```

Comment:
Just a quick answer before leaving home...yes, typically.we use TransactionBuilder to perform a series of read/write actions.

TransactionBuilders are short lived and cannot be used anymore after queueing their result. But you can also use the BtLEQueue directly, e.g. getQueue().add(new WriteAction(...))

Might be that we need to add a insert() or addFront() method to BtLEQueue so that it will execute the given action right away instead of adding it to the end of the list.